src/test/java/com/google/cloud/compute/v1/it/ITInstanceTest.java

```diff
@@ -42,42 +79,225 @@ public static void setUp() throws IOException {
+    assertThat(instance.getName()).isEqualTo(INSTANCE);
+    assertThat(instance.getDeletionProtection()).isFalse();
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void setLabelsInstanceTest() {
+    String key1 = "key-" + UUID.randomUUID().toString();
+    String key2 = "key-" + UUID.randomUUID().toString();
+    Map<String, String> labels = ImmutableMap.of(key1, "label-1", key2, "label-2");
+    InstancesSetLabelsRequest instancesSetLabelsRequestResource =
+        InstancesSetLabelsRequest.newBuilder()
+            .setLabelFingerprint(getInstance().getLabelFingerprint())
+            .putAllLabels(labels)
+            .build();
+    Operation operation =
+        instanceClient.setLabelsInstance(FORMATTED_INSTANCE, instancesSetLabelsRequestResource);
+    assertOperationDetails(operation, "setLabels");
+    assertThat(getInstance().getLabelFingerprint()).isNotNull();
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void setMachineResourcesInstanceTest() {
+    InstancesSetMachineResourcesRequest instancesSetMachineResourcesRequestResource =
+        InstancesSetMachineResourcesRequest.getDefaultInstance();
+    Operation operation =
+        instanceClient.setMachineResourcesInstance(
+            FORMATTED_INSTANCE, instancesSetMachineResourcesRequestResource);
+    assertOperationDetails(operation, "setMachineResources");
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void setMachineTypeInstanceTest() {
+    InstancesSetMachineTypeRequest instancesSetMachineTypeRequestResource =
+        InstancesSetMachineTypeRequest.newBuilder().setMachineType(MACHINE_TYPE).build();
+    Operation operation =
+        instanceClient.setMachineTypeInstance(
+            FORMATTED_INSTANCE, instancesSetMachineTypeRequestResource);
+    assertOperationDetails(operation, "setMachineType");
+    assertThat(getInstance().getMachineType().contains("machineTypes/n1-standard-1")).isTrue();
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void setMinCpuPlatformInstanceTest() {
+    String minCpuPlatform = "Intel Haswell";
+    InstancesSetMinCpuPlatformRequest instancesSetMinCpuPlatformRequestResource =
+        InstancesSetMinCpuPlatformRequest.newBuilder().setMinCpuPlatform(minCpuPlatform).build();
+    Operation operation =
+        instanceClient.setMinCpuPlatformInstance(
+            FORMATTED_INSTANCE, instancesSetMinCpuPlatformRequestResource);
+    assertOperationDetails(operation, "setMinCpuPlatform");
+    assertThat(getInstance().getCpuPlatform()).isEqualTo(minCpuPlatform);
+  }
+
+  private Instance getInstance() {
+    GetInstanceHttpRequest httpRequest =
+        GetInstanceHttpRequest.newBuilder().setInstance(FORMATTED_INSTANCE).build();
+    return instanceClient.getInstance(httpRequest);
+  }
+
+  private void waitUntilStatusChangeTo(String status) {
+    while (true) {
+      Instance instance = getInstance();
+      if (instance.getStatus().equals(status)) {
+        break;
+      }
     }
```

Comment:
this will hang if the status is never changed to expected.
It's better to use:
```
long expirationTime = System.getCurrentTimeMillis() + timeout
...
if (instance.getStatus().equals(status) || System.getCurrentTimeMillis() > expirationTime ) {
    break;
}
...
assertThat(instance.getStatus().equals(status))
```

Reply:
don't worry it will change as expected